pandas/tests/io/formats/test_to_csv.py

```diff
@@ -563,3 +563,23 @@ def test_to_csv_compression(self, compression_only,
             result = pd.read_csv(path, index_col=0,
                                  compression=read_compression)
             tm.assert_frame_equal(result, df)
+
+
+
+def test_csv_formatter_line_terminator_default(monkeypatch):
+    # see GH #23608
+    # ensure default line_terminator used is os.linesep
+    df = tm.makeDataFrame()
+
+    with monkeypatch.context() as m:
+        # fake default os.linesep to look like windows
+        m.setattr(os, 'linesep', '\r\n')
+        assert os.linesep == '\r\n'
+        # import within this context so it uses the patched os.linesep
+        from pandas.io.formats.csvs import CSVFormatter
```

Comment:
Hmm does this import need to actually happen here as well? Not terribly familiar with the codebase of it but somewhat surprised it's required

Reply:
I don't have a full understanding of how scoping imports works in Python tbh. I'm trying to do two things: 1. Make sure CSVFormatter uses the spy version of os.linesep 2. Avoid conflicts with other tests that use os.linesep.